standard.md

```diff
@@ -0,0 +1,94 @@
+- A list of the options defined by the diagnostic to configure it. It is recommended that this list comes with a small explanation of it's meaning and effects, the type of the expected values and any default values taken when they are not explivitly defined (recommended).
+- A list of the mandatory common options required for the diagnostic to work (recommended).
+- A link to the online documentation, if available (suggested)
+
+## The diagnostic configuration file
+
+This section will define the syntax fot he main configuration file. It must be a yaml file containing a dictionary that contains a dictionary with all the options needed by a diagnostic
+
+### Required options
+
+The following options must be present in the configuration file passed to the diagnostic.
+
+- **input_files**: `list`. List of absolute paths to the metadata files describing the data to be used by the diagnostic. The format of this files is defined later.
+- **tool**: `str`. Name of the tool used to prepare the data for the diagnostic.
+- **version**: `str` Version of the tool used to prepare the data for the diagnostic.
+- **run_dir**: `str`. Path to the folder to use as current path for diagnostic execution. Must be writable by the diagnostic.
+- **data_dir**: `str`. Path to the folder to store the  diagnostic's generated data on. Must be writable by the diagnostic.
+- **plot_dir**: `str`. Path to the folder to store the  diagnostic's generated figures on. Must be writable by the diagnostic.
+
+It is suggested to provide different folders for `run_dir`, `data_dir` and `plot_dir`, as it will be easier for the users to interact with the results, but the same folder can be used.
+
+### Reserved options
+
+The options listed in this file are not required for the diagnostics to implement, but if they do they must follow the following guidelines:
+
+- **write_plots**: `boolean`, default value is `true`. A flag to control if the diagnostic generates the associated figures with the results or not.
+- **write_data**: `boolean`, default value is `true`. A flag to control if the diagnostic generates the associated data files with the results or not.
+- **log_level**: `str`, default values is `info`. Sets the granularity of the diagnostic logs and console output. If used, diagnostic must accept at least `error` `warning`, `info` and `debug`
+
+
+## The data definition file
+
+The data definition file must be a yaml file containing a dictionary of dictionaries. This first dictionary will have as keys the path to the data files and as values the dictionary containg the metadata of each file.
+
+There is no limit to the number of file that can be provided on a single file, although it is suggested
+
+### Required options
+
+- **alias**: `str`. Unique name for the dataset. Any (`alias`, `variable`) pair must be unique
+- **filename**: `str`. Absoulute path to the data file.
+- **variable**: `str`. Variable name to be used by the diagnostic.
```

Comment:
This is a quirck of ESMValTool that sometimes is useful. It is mostly used to get two instances of the same variable that have been preprocessed differently.

For example, if you want to produce an index made of the difference of the tas mean in two boxes, you can have two vars `tas-box1` `tas-box2` that are in fact tas in two different places.

We do not change the name in the file, though, so we need to keep track of both. I am not sure if this is good behaviour or not, I can see arguments for both

Reply:
This raises a more basic question : how do the tool and the diag script agree on which data file produced by the tool matches  which data file expected by the diag. This is quite ambiguous in ESMValTool and in the current draft. 

I understand here that, as a basis, the `short_name` is used for that, and,  if ambiguous, the `variable` is then used. But it is a bit tricky.

I would prefer that the match is ensured by ordering the input files listed in the main yaml file (this would also allow scripts to avoid having to dig in input files for finding which is what)

The other possibility is that we rename `variable` in `input_field_id` to better show that this attribute is used for the match. This implies that diags doc must list the input_field_ids they use, and describe the corresponding expected inputs